specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/componentProactiveDetection_API.json

```diff
@@ -0,0 +1,312 @@
+              }
+            }
+          },
+          "x-ms-examples": {
+            "ProactiveDetectionConfigurationsList": {
+              "$ref": "./examples/ProactiveDetectionConfigurationsList.json"
+            }
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/ProactiveDetectionConfigs/{ConfigurationId}": {
+        "delete": {
+          "description": "Delete a ProactiveDetection configuration of an Application Insights component.",
+          "operationId": "ProactiveDetectionConfigurations_Delete",
+          "parameters": [
+            {
+              "$ref": "#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            },
+            {
+              "$ref": "#/parameters/SubscriptionIdParameter"
+            },
+            {
+              "$ref": "#/parameters/ResourceNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ConfigurationId"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "The ProactiveDetection configuration that was successfully deleted."
+            }
+          },
+          "x-ms-examples": {
+            "ProactiveDetectionConfigurationDelete": {
+              "$ref": "./examples/ProactiveDetectionConfigurationDelete.json"
+            }
+          }
+        },
+        "get": {
+          "description": "Get the ProactiveDetection configuration for this configuration id.",
+          "operationId": "ProactiveDetectionConfigurations_Get",
+          "parameters": [
+            {
+              "$ref": "#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            },
+            {
+              "$ref": "#/parameters/SubscriptionIdParameter"
+            },
+            {
+              "$ref": "#/parameters/ResourceNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ConfigurationId"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "The ProactiveDetection configuration for this configuration id.",
+              "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentProactiveDetectionConfiguration"
+              }
+            }
+          },
+          "x-ms-examples": {
+            "ProactiveDetectionConfigurationGet": {
+              "$ref": "./examples/ProactiveDetectionConfigurationGet.json"
+            }
+          }
+        },
+        "put": {
+          "description": "Update the ProactiveDetection configuration for this configuration id.",
+          "operationId": "ProactiveDetectionConfigurations_Update",
```

Comment:
Is this only for updating or is it also for creating?  If it can do both then the operation ID should be "ProactiveDetectionConfigurations_CreateOrUpdate".

Reply:
Logically – it’s updating (since there is a default conf for everyone, which also appears in UI in case there is no configuration in the DB).

Physically – we create a conf item only when it is changed to be different then the default one.
Then, I think it’s an implementation detail, and “Update” is enough. WDYT?

Reply:
If you can specify values for ResourceNameParameter and ConfigurationId (I'm guessing these are the params of interest, please correct me if I'm wrong) that don't already exist and a new "thing" is created then I think it needs to be CreateOrUpdate.

Reply:
I think it might be confusing, since it causes the user to think he creates something which doesn't exist (i.e - without creating it, we won't do anything).
We have about 10 algorithms we run. Each algorithm has a different ConfigurationId (total of about 10 configuration Ids currently. a new configuration Id is introduces only when we implement a new algorithm).

When we run our processing for a specific customer, we iterate over these 10 types, if the user doesn't have a configuration for this configurationId  - we follow the default behavior -
1. Run the algorithm (Enabled)
2. Send the results to a default email address. 
If the user has a configuration (i.e - he performed a put operation before)- 
1. Run the algorithm only if it is enabled.
2. Send the email to specified customers in the configuration.

Do you still think it should be CreateOrUpdate?

Reply:
Thanks for the explanation, I agree it should be left as Update.  Regarding the delete API, is that something that can be invoked via the UI?  If so then I think it should be included in the swagger to keep the behavior at parity.

Reply:
No, it can't be invoked from UI.
In the UI we use only GET and PUT.

Not sure why is it exposed. perhaps for some cleanup processes in the service which manages all the configurations. ProactiveDetection UI and BE do not use the Delete option in the API.